infra/aws/terraform/registry.k8s.io/s3/bucket.tf

```diff
@@ -64,3 +64,30 @@ resource "aws_s3_bucket_ownership_controls" "registry-k8s-io" {
     aws_s3_bucket_policy.registry-k8s-io-public-read
   ]
 }
+
+resource "aws_iam_user_policy" "registry-k8s-io-rw" {
+  name = "${aws_s3_bucket.registry-k8s-io.bucket}-access"
+  user = var.iam_user_name
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : [
+          "s3:ListBucket"
+        ],
+        "Effect" : "Allow",
+        "Resource" : "${aws_s3_bucket.registry-k8s-io.arn}/"
+      },
+      {
+        "Action" : [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject"
+        ],
+        "Effect" : "Allow",
+        "Resource" : "${aws_s3_bucket.registry-k8s-io.arn}/*"
+      }
+    ]
+  })
```

Comment:
The _registry.k8s.io_s3writer_ Role is currently only used for provisioning the buckets and bucket permissions. I think this role is best for managing the infra. I think that it would be best to have a role or user policy that can only read/write to the buckets.

Reply:
> I'm wary of defining an access key via Terraform.
> 
>     * How will we ensure that the credentials for this IAM user are secure?
> 
>     * How will will ensure that credential rotation happens?
> 
>     * Who or what will have access to the secret key for this IAM user?
> 
>     * What additional controls might be needed? (eg multifactor authentication; two-person approval for changes; enforced use of TLS; additional detective controls)
> 
>     * Who has access to change this user's credentials, and how do we assure this?


@sftim I think your questions are more related to the overall security aspect of the AWS org owning those buckets and deserve to be their own Github issues. Feel free to open them on https://github.com/cncf-infra/aws-infra/.

Reply:
Note that this isn't a role... it's an IAM User and Policy, which is different from a role... and unless I'm mistaken, a separate IAM User and IAM User Policy is going to be created in each AWS region that will only have the ability to list and write to the bucket in that specific region. Is that correct?

If that is the case, then we're going to need to have separate IAM access keys stored in secrets manager for each of those 11 AWS regions... right?